src/ram/analysis/Index.cpp

```diff
@@ -396,73 +414,31 @@ void IndexAnalysis::run(const TranslationUnit& translationUnit) {
         // Currently RAM does not have such situation.
         const std::string& relA = swap.getFirstRelation();
         const std::string& relB = swap.getSecondRelation();
-        MinIndexSelection& indexesA = getIndexes(relA);
-        MinIndexSelection& indexesB = getIndexes(relB);
-        // Add all searchSignature of A into B
-        for (const auto& signature : indexesA.getSearches()) {
-            indexesB.addSearch(signature);
-        }
 
-        // Add all searchSignature of B into A
-        for (const auto& signature : indexesB.getSearches()) {
-            indexesA.addSearch(signature);
-        }
-    });
+        const auto searchesA = relationToSearches[relA];
+        const auto searchesB = relationToSearches[relB];
 
-    // find optimal indexes for relations
-    for (auto& cur : minIndexCover) {
-        MinIndexSelection& indexes = cur.second;
-        indexes.solve();
-    }
+        relationToSearches[relA].insert(searchesB.begin(), searchesB.end());
+        relationToSearches[relB].insert(searchesA.begin(), searchesA.end());
+    });
 
-    // Only case where indexSet is still empty is when relation has arity == 0
-    for (auto& cur : minIndexCover) {
-        MinIndexSelection& indexes = cur.second;
-        if (indexes.getAllOrders().empty()) {
-            indexes.insertDefaultTotalIndex(0);
+    // remove all empty searches
+    for (auto& relToSearch : relationToSearches) {
+        auto& searches = relToSearch.second;
+        for (auto it = searches.begin(); it != searches.end();) {
+            if (it->empty()) {
+                it = searches.erase(it);
+            } else {
+                ++it;
+            }
         }
     }
-}
-
-MinIndexSelection& IndexAnalysis::getIndexes(const std::string& relName) {
-    auto pos = minIndexCover.find(relName);
-    if (pos != minIndexCover.end()) {
-        return pos->second;
-    } else {
-        auto ret = minIndexCover.insert(std::make_pair(relName, MinIndexSelection()));
-        assert(ret.second);
-        return ret.first->second;
-    }
-}
-
-void IndexAnalysis::print(std::ostream& os) const {
```

Comment:
Can you leave a basic print feature in the analysis? This is important to understand which indexes were selected and can be retrieved from the debug report.